packages/nimble-components/src/wafer-map/modules/hover-handler.ts

```diff
@@ -0,0 +1,88 @@
+import type { WaferMap } from '..';
+import { PointCoordinates, WaferMapQuadrant } from '../types';
+
+/**
+ * HoverHandler deals with user interactions and events like hovering
+ */
+export class HoverHandler {
+    public constructor(private readonly wafermap: WaferMap) {
+        wafermap.hoverWidth = wafermap.dataManager!.dieDimensions.width * wafermap.transform.k;
+        wafermap.hoverHeight = wafermap.dataManager!.dieDimensions.height * wafermap.transform.k;
+    }
+
+    public mousemove(event: MouseEvent): void {
+        const mousePosition: PointCoordinates = {
+            x: event.offsetX,
+            y: event.offsetY
+        };
+
+        if (!this.hoversOverDie(mousePosition)) {
+            this.wafermap.hoverDie = undefined;
+            return;
+        }
+
+        // get original mouse position in case we are in zoom.
+        const invertedPoint = this.wafermap.transform.invert([
+            mousePosition.x,
+            mousePosition.y
+        ]);
+
+        const dieCoordinates = this.calculateDieCoordinates({
+            x: invertedPoint[0],
+            y: invertedPoint[1]
+        });
+
+        this.wafermap.hoverDie = this.wafermap.dataManager!.getWaferMapDie(dieCoordinates);
+    }
+
+    public mouseout(_event: MouseEvent): void {
+        this.wafermap.hoverDie = undefined;
+    }
+
+    private calculateDieCoordinates(
+        mousePosition: PointCoordinates
+    ): PointCoordinates {
+        const axisLocation = this.wafermap.quadrant;
+        const xRoundFunction = axisLocation === WaferMapQuadrant.bottomLeft
+            || axisLocation === WaferMapQuadrant.topLeft
+            ? Math.floor
+            : Math.ceil;
+        const yRoundFunction = axisLocation === WaferMapQuadrant.topLeft
+            || axisLocation === WaferMapQuadrant.topRight
+            ? Math.floor
+            : Math.ceil;
+        // go to x and y scale to get the x,y values of the die.
+        const x = xRoundFunction(
+            this.wafermap.dataManager!.horizontalScale.invert(
+                mousePosition.x - this.wafermap.dataManager!.margin.left
+            )
+        );
+        const y = yRoundFunction(
+            this.wafermap.dataManager!.verticalScale.invert(
+                mousePosition.y - this.wafermap.dataManager!.margin.top
+            )
+        );
+        return { x, y };
+    }
+
+    private hoversOverDie(mousePosition: PointCoordinates): boolean {
+        const canvasContext = this.wafermap.canvas.getContext('2d', {
+            willReadFrequently: true
+        });
```

Comment:
I don't think this actually works as expected. The wafer map is requesting the canvas context a lot and with different parameters. I think instead we need to request it once and save the context to re-use everywhere:

```ts
// index.ts

/* @internal */
public canvasContext!;
connectedCallback () {
   this.canvasContext = this.canvas.getContext('2d',{willReadFrequently: true});
}
```

It does not need to be in this PR but should be in a follow-up.

See the following discussion: https://stackoverflow.com/a/74136040
>  I found in my case, I was creating the canvas and writing to it in one function, then returning the canvas. Later in my code, the result of that function call took the same canvas and created another context. It applied the { willReadFrequently: true } options argument to that call, but that was the second time getContext was called for this canvas. That means the texture buffer was already living on the GPU by this point and the second getContext call was ignoring the willReadFrequently suggestion as the data was already on the GPU.